`make` doesn't work for a `tools` directory

Figure out why `make` doesn't work for a top level `tools` directory and fix it:

```
vagrant@buffy-leader-1:~/wallaroo/tools$ make 
sudo: cset: command not found
make: *** No rule to make target 'build-tools-all', needed by 'build'.  Stop.
```

--- a/rules.mk
+++ b/rules.mk
@@ -403,8 +403,9 @@ endef
 # rule to generate includes for makefiles in subdirs of first argument
 define make-goal
 $(eval MAKEDIRS := $(sort $(dir $(wildcard $(1:%/=%)/*/Makefile))))
+$(eval MAKEFILES := $(sort $(wildcard $(1:%/=%)/*/Makefile)))
 $(foreach mdir,$(MAKEDIRS),$(eval $(notdir $(mdir:%/=%)) := $(mdir)))
-$(foreach mdir,$(MAKEDIRS),$(eval include $($(notdir $(mdir:%/=%)))Makefile))
+$(eval include $(MAKEFILES))
 endef
 
 # rule to generate targets for building actual pony executable including dependencies to relevant *.pony files so incremental builds work properly

